sc2/bot_ai.py

```diff
@@ -209,7 +217,14 @@ def _correct_zerg_supply(self):
         await self.build(building, near=location, max_distance=max_distance, random_alternative=False, placement_step=1)
 
     async def get_next_expansion(self) -> Optional[Point2]:
-        """Find next expansion location."""
+        """Find next expansion location.
+        Example usage:
+        self.actions = []
+        next_expansion_location = await self.get_next_expansion()
+        self.actions.append(self.workers.closest_to(next_expansion_location).build(HATCHERY, next_expansion_location))
+        if self.actions:
+            await self.do_actions(self.actions)
```

Comment:
This example looks strange to me
- `get_next_expansion()` can return `None` in which case `closest_to` will crash
- `self.actions` will always be non-empty (if you get that far), so the `if` is unnecessary

I would rewrite it as
```python
next_loc = await self.get_next_expansion()
if next_loc is not None:
  await self.do_actions([self.workers.closest_to(next_loc).build(HATCHERY, next_loc)])
```
I guess the point is that you shouldn't call `do_actions` on individual actions but collect them in a list and submit them all at once, but that message isn't really clear from the example. Maybe add `# ... more actions ...` before and after adding the `build` action?

Reply:
Yeah, that was the point, Ill make it more clear